zipline-loader/src/main/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -0,0 +1,156 @@
+package app.cash.zipline.loader
+
+import app.cash.zipline.Zipline
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.Semaphore
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.sync.withPermit
+import okio.ByteString
+
+/**
+ * Gets code from an HTTP server or a local cache,
+ * and loads it into a zipline instance. This attempts
+ * to load code as quickly as possible, and will
+ * concurrently download and load code.
+ */
+class ZiplineLoader(
+// TODO add caching
+//  val cacheDirectory: Path,
+//  val cacheMaxSizeInBytes: Int = 100 * 1024 * 1024,
+  val client: ZiplineHttpClient,
+) {
+  /*
+  file a.js:
```

Comment:
Move this into the test case? Your test’s fake HTTP client can return a string like this to serve as a sample payload